tico/experimental/quantization/ptq/observers/base.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import torch
+
+from tico.experimental.quantization.ptq.dtypes import DType, UINT8
+from tico.experimental.quantization.ptq.qscheme import QScheme
+
+
+class ObserverBase(ABC):
```

Comment:
I think algorithm-specific implementations are still included in this class. For example,

channel_axis,
min_val, max_val
compute_qparams,
fake_quant

The above fields and methods are for min/max-based affine quantization. Could you pull them down to MinMaxObserver? ObserverBase would be like

ObserverBase
-> `__init__()`
-> `reset` (abstract method -> Or we can make this as a final method and define another abstract method for algorithm-specific implementation. (see strategy pattern https://refactoring.guru/design-patterns/strategy) )
-> `load_qparams`
-> `has_qparams`
-> `collect` (abstrat method)
-> `compute_qparams` (abstrat method)
-> `fake_quant` (abstrat method)
-> `__repr__` (abstrat method)

Reply:
@jinevening Well, I think most practical observers like MinMax, Percentile, KL/Histogram, etc reduce to producing [min_val, max_val] and compute quantization parameter accordingly. Other deviations can be overridden. 

Are you considering any specific observer that can't match with this base class?

Reply:
Got it. This is what I'm gonna do.

1. Make `ObserverBase` slim.
2. Introduce `AffineObserverBase` that inherits `ObserverBase` for current observers in the draft having common logic. Current observer now inherits this class.
3. Introduce `MXObserver` that doesn't track min/max or produce affine qparams as an example of a non-affine observer.